src/ivectorbin/ivector-plda-scoring-dense.cc

```diff
@@ -0,0 +1,226 @@
+      scores_wspecifier = po.GetArg(4);
+
+    Plda plda;
+    ReadKaldiObject(plda_rxfilename, &plda);
+
+    SequentialTokenVectorReader spk2utt_reader(spk2utt_rspecifier);
+    RandomAccessBaseFloatVectorReader ivector_reader(ivector_rspecifier);
+    BaseFloatMatrixWriter scores_writer(scores_wspecifier);
+    int32 num_spk_err = 0,
+          num_spk_done = 0;
+    for (; !spk2utt_reader.Done(); spk2utt_reader.Next()) {
+      Plda this_plda(plda);
+      std::string spk = spk2utt_reader.Key();
+
+      // The uttlist is sorted here and in binaries that use the scores
+      // this outputs.  This is to ensure that the segment corresponding
+      // to the same rows and columns (of the score matrix) across binaries.
+      std::vector<std::string> uttlist = spk2utt_reader.Value();
+      std::sort(uttlist.begin(), uttlist.end());
+      std::vector<Vector<BaseFloat> > ivectors;
+
+      for (size_t i = 0; i < uttlist.size(); i++) {
+        std::string utt = uttlist[i];
+
+        if (!ivector_reader.HasKey(utt)) {
+          KALDI_ERR << "No iVector present in input for utterance " << utt;
+        }
+
+        Vector<BaseFloat> ivector = ivector_reader.Value(utt);
+        ivectors.push_back(ivector);
+      }
+      if (ivectors.size() == 0) {
+        KALDI_WARN << "Not producing output for recording " << spk
+                   << " since no segments had iVectors";
+        num_spk_err++;
+      } else {
+        Matrix<BaseFloat> ivector_mat(ivectors.size(), ivectors[0].Dim()),
+                          ivector_mat_pca,
+                          ivector_mat_plda,
+                          pca_transform,
+                          scores(ivectors.size(), ivectors.size());
+
+        for (size_t i = 0; i < ivectors.size(); i++) {
+          ivector_mat.Row(i).CopyFromVec(ivectors[i]);
+        }
+        if (EstPca(ivector_mat, target_energy, &pca_transform)) {
+          // Apply PCA transform to the raw i-vectors.
+          ApplyPca(ivector_mat, pca_transform, &ivector_mat_pca);
+
+          // Apply PCA transform to the parameters of the PLDA model.
+          this_plda.ApplyTransform(Matrix<double>(pca_transform));
+
+          // Now transform the i-vectors using the reduced PLDA model.
+          TransformIvectors(ivector_mat_pca, plda_config, &this_plda,
+            &ivector_mat_plda);
+        } else {
+          KALDI_WARN << "Unable to compute conversation dependent PCA for"
+            << " recording " << spk << ".";
+          ivector_mat_pca.Resize(ivector_mat.NumRows(), ivector_mat.NumCols());
+          ivector_mat_pca.CopyFromMat(ivector_mat);
+        }
+        for (int32 i = 0; i < ivector_mat_plda.NumRows(); i++) {
+          for (int32 j = 0; j < ivector_mat_plda.NumRows(); j++) {
+            // Pass the raw PLDA scores through a logistic function
+            // so that they are between 0 and 1.
+            scores(i,j) = 1.0
+              / (1.0 + exp(this_plda.LogLikelihoodRatio(Vector<double>(
```

Comment:
@mmaciej2 did you ever determine if passing this through the logistic function was necessary? Would it be preferable to use the raw scores?